frontend/src/metabase/dashboard/containers/DashboardHeader.styled.tsx

```diff
@@ -26,9 +31,18 @@ export const DashboardHeaderButton = styled(Button)<{ isActive?: boolean }>`
   svg {
     vertical-align: middle;
   }
+
+  ${breakpointMaxSmall} {
+    ${props =>
+      css`
+        ${props.visible ? "" : "display: none;"}
```

Comment:
We usually don't render the element at all rather than hiding it in CSS. Can you do that? Then you only need to change the E2E test to `should.not.exist` I think.

Reply:
I tried this initially and it turned out we have a pretty complicated logic to determine whether to render some button or not. So I came up to understanding we shouldn't change any logic in the code, instead we should just hide the button for small screens and it's straightforward with css. If something is doable in css, why do we need to bother js? 😄

Reply:
> If something is doable in css, why do we need to bother js?

Good point. This might be worth another FE discussion. At least in many places, for the case of trying to render vs hide something we always try to not render it. I'd just want to make things consistent. I'm not totally sure why you say it's complicated.

Can we try using `useSmallScreen` 
https://github.com/metabase/metabase/blob/fab35e5b5b01cf3410672e4adf829fe7dfed2de3/frontend/src/metabase/hooks/use-is-small-screen.ts#L3

Example usage
https://github.com/metabase/metabase/blob/fab35e5b5b01cf3410672e4adf829fe7dfed2de3/frontend/src/metabase/nav/components/AppBar/AppBar.tsx#L26-L37

CC. @ranquild since I remembered we've talked about using CSS to hide elements vs not rendering elements before and I think the decision was that we want to not render elements instead.

Reply:
Also, just in case we go the CSS route. I don't think we should pass `visible={false}` that could be misleading as it's only applicable to small screen size. You might better just use media query and directly e.g.
```ts
${breakpointMaxSmall} {
  display: none;
}
```

Reply:
we render all the buttons with the same component, so adding this rule will hide all icons on small screens @WiNloSt

Reply:
renamed variable to `visibleOnSmallScreen`

Reply:
Gotcha why we need `visible={false}`